middle_end/flambda2/terms/check_attribute.ml

```diff
@@ -0,0 +1,41 @@
+module Property = struct
+  type t =
+    | Noalloc
+
+  let to_string  = function
+    | Noalloc -> "noalloc"
+
+  let equal x y =
+    match x, y with
+    | Noalloc, Noalloc -> true
+
+  let from_lambda : Lambda.property -> t  = function
+    | Noalloc -> Noalloc
+end
+
+type t =
+  | Default_check
+  | Assert of Property.t
+  | Assume of Property.t
+
+let print ppf t =
+  match t with
+  | Default_check -> ()
+  | Assert p -> Format.fprintf ppf "assert %s@ " (Property.to_string p)
+  | Assume p -> Format.fprintf ppf "assume %s@ " (Property.to_string p)
```

Comment:
I think these should be e.g. `@[(assume %s)@]`

Reply:
> I think these should be e.g. `@[(assume %s)@]`

Fixed, thanks! 
Tested with `-dflambda` compilation flag which prints out  either `(assume noalloc)` or `(assert noalloc)` when attributes are present, and a grayed out `()` without annotations, i.e., when the attribute is set to `Default_check`.